src/vs/workbench/contrib/extensions/browser/extensionsWorkbenchService.ts

```diff
@@ -884,7 +893,11 @@ export class ExtensionsWorkbenchService extends Disposable implements IExtension
 			return Promise.resolve();
 		}
 
-		const toUpdate = this.outdated.filter(e => !this.isAutoUpdateIgnored(new ExtensionIdentifierWithVersion(e.identifier, e.version)));
+		let toUpdate = this.outdated.filter(e => !this.isAutoUpdateIgnored(new ExtensionIdentifierWithVersion(e.identifier, e.version)));
+		if (!this.isAutoUpdateDisabledExtensionsEnabled()) {
+			toUpdate = toUpdate.filter(e => e.local && this.extensionEnablementService.isEnabled(e.local));
+		}
```

Comment:
Apply the filtering once instead of iterating twice. Also only exclude globally disabled extensions. Use following check for that 

```
e.enablementState === EnablementState.DisabledGlobally
```